runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -1806,6 +1806,17 @@ handleServerMessage(JITServer::ClientStream *client, TR_J9VM *fe, JITServer::Mes
                   if (ramMethod) createMethod = true;
                   break;
                   }
+               case TR_ResolvedMethodType::ImproperInterface:
+                  {
+                  TR::VMAccessCriticalSection getResolvedHandleMethod(fe);
+                  ramMethod = jitGetImproperInterfaceMethodFromCP(
+                     fe->vmThread(),
+                     owningMethod->cp(),
+                     cpIndex,
+                     reinterpret_cast<UDATA *>(&vTableOffset));
```

Comment:
`jitGetImproperInterfaceMethodFromCP` takes `vTableOffset` as a pointer to `UDATA`, but the local above is declared to be `uint32_t vTableOffset = 0;`
I think this is dangerous in that `jitGetImproperInterfaceMethodFromCP` may copy a 64-bit entity and silently corrupt the stack around ` vTableOffset` local.
We need to declare `vTableOffset` as `UDATA` and if we really have to, cast that into a `uint32_t` when storing into `vTableOffsets` vector (assuming the offset is never exceeding 32-bit)

Reply:
I implemented what you said above. On my machine it compiles without casting to `uint32_t` but I'm not certain that it will work on all platforms.
In the below example it's simply cast to `uint32_t`:
https://github.com/eclipse/openj9/blob/e006f8232bb7f1273c776a451ea8330bd7f224bb/runtime/compiler/env/j9method.cpp#L6491

Reply:
Could you please repeat the same trick for `TR_ResolvedJ9Method::getVirtualMethod` above?

Reply:
Done.